docs/website/01_framework/controlRoutines.md

```diff
@@ -167,3 +173,23 @@ struct StandardArcLength {
     std::optional<double> psi;
   };
 ```
+
+## Automatic step-sizing for the path-following techniques
+
+Instead of using a constant step size, the step size can be automatically adapted for efficient computations.
+The automatic step-sizing is implemented according to Ramm[@wunderlich_strategies_1981].
+The step size can be scaled as shown below:
+
+$$
+\hat{s}_{k+1} = \left(\frac{\hat{i}}{i_k}\right)^{1/2}\hat{s}_k
+$$
+
+Here, $\hat{s}_{k+1}$ and $\hat{s}_{k}$ are the step sizes at $k+1$-th and $k$-th iteration.
+Here, $\hat{i}$ is the desired number of iterations and $i_k$ is the number of iterations used in the previous step.
+
+For implementation details, refer to `ikarus/controlroutines/adaptivestepsizing.hh`.
+
+By default, `AdaptiveStepSizing::NoOp` is used with a path-following technique.
```

Comment:
It would be better to explain the interface that theses adaptive step sizing routines have to follow. The current comments are a bit useless. You explain that NoOp is doing nothing with some argument.
Then you comment on the IterationBased again.

I think this section should be as follows. Explain the interface of these functions. and then talk about these two implementations. and first start with the default namely doing nothing

Reply:
Is it better now? What do you think? @rath3t

Reply:
As discussed, the documentation of concepts will be separated for better readability in a separate PR. Refer #217